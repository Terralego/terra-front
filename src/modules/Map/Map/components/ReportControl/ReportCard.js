import React, { useState, useCallback } from 'react';
import { Overlay, Card, Elevation } from '@blueprintjs/core';

import ReportForm from './ReportForm';
import ReportSuccess from './ReportSuccess';

export const ReportCard = ({
  isOpen,
  onSubmit,
  cancelReport,
  coordinates,
  reportUrl,
  newReport,
  endReport,
  translate: t,
}) => {
  const [submited, setSubmited] = useState(false);

  const onNewReport = useCallback(() => {
    setSubmited(false);
    newReport();
  }, [newReport]);

  const submitReport = useCallback(value => {
    setSubmited(true);
    onSubmit(value);
  }, [onSubmit]);


  return (
    <Overlay
      autoFocus
      lazy
      canOutsideClickClose
      canEscapeKeyClose
      isOpen={isOpen}
      onClose={cancelReport}
    >
      <Card
        interactive
        elevation={Elevation.FOUR}
        className="report-card"
      >
        {(submited && <ReportSuccess newReport={onNewReport} endReport={endReport} translate={t} />)
          || (
            <ReportForm
              onSubmit={submitReport}
              onCancel={cancelReport}
              coordinates={coordinates}
              url={reportUrl}
              translate={t}
            />
          )}
      </Card>
    </Overlay>
  );
};

export default ReportCard;
