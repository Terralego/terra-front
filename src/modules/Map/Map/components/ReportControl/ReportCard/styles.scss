.overlay-container {
    width: 100vw;
    height: 100vh;
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .report-card {
    height: auto;
    width: auto;
  }
