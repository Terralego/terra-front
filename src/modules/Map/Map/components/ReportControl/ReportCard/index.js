import React, { useState } from 'react';
import { Overlay, Card, Elevation } from '@blueprintjs/core';

import ReportForm from '../ReportForm';
import ReportSuccess from '../ReportSuccess';

import './styles.scss';

export const ReportCard = ({
  isOpen,
  onSubmit,
  cancelReport,
  coordinates,
  reportUrl,
  newReport,
  endReport,
  translate: t,
}) => {
  const [submited, setSubmited] = useState(false);

  const title = submited ?
    'terralego.map.report_control.success.title' :
    'terralego.map.report_control.form.title';

  const onNewReport = () => {
    setSubmited(false);
    newReport();
  };

  const onEndReport = () => {
    setSubmited(false);
    endReport();
  };

  const submitReport = data => {
    setSubmited(true);
    onSubmit(data);
  };


  return (
    <Overlay
      autoFocus
      lazy
      canEscapeKeyClose
      isOpen={isOpen}
      onClose={cancelReport}
      className="overlay-container"
    >
      <Card
        interactive
        elevation={Elevation.FOUR}
        className="report-card"
      >
        <h1>{t(title)}</h1>
        {submited
          && <ReportSuccess newReport={onNewReport} endReport={onEndReport} translate={t} />}

        {!submited
        && (
        <ReportForm
          onSubmit={submitReport}
          onCancel={cancelReport}
          coordinates={coordinates}
          url={reportUrl}
          translate={t}
        />
        )}
      </Card>
    </Overlay>
  );
};

export default ReportCard;
