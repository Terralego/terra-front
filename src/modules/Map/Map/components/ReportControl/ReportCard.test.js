import React from 'react';
import renderer from 'react-test-renderer';

import ReportCard from './ReportCard';

jest.mock('@blueprintjs/core', () => ({
  Overlay: ({ children }) => <div id="overlay">{children}</div>,
  Card: ({ children }) => <div id="card">{children}</div>,
  Elevation: () => ({ FOUR: 'four' }),
}));
jest.mock('./ReportForm', () => () => 'reportform');
jest.mock('./ReportSuccess', () => () => 'reportsuccess');

it('should render correctly', () => {
  const tree = renderer.create(
    <ReportCard
      isOpen
      reportUrl="report/url/"
      translate={jest.fn()}
      cancelReport={jest.fn()}
      onSubmit={jest.fn()}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
