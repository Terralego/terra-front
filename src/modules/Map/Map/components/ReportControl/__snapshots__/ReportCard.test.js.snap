// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  id="overlay"
>
  <div
    id="card"
  >
    reportform
  </div>
</div>
`;
